### PR TITLE
Vmware virtustream

### PR DESCRIPTION
**Pull request by VeryEarly**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/VMWare/docs/Remove-AzVMWareAuthorization.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+Remove-AzVMWareAuthorization -InputObject <IVMWareIdentity> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait]
+ [-PassThru] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Delete an ExpressRoute Circuit Authorization in a private cloud
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Remove-AzVMWareAuthorization.md:33]: Here miss the examples.

File: src/VMWare/docs/Remove-AzVMWareCluster.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+### DeleteViaIdentity
+```
+Remove-AzVMWareCluster -InputObject <IVMWareIdentity> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait]
+ [-PassThru] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Delete a cluster in a private cloud
+
+## EXAMPLES
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Remove-AzVMWareCluster.md:31]: Here miss the examples.

File: src/VMWare/docs/Remove-AzVMWarePrivateCloud.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+Remove-AzVMWarePrivateCloud -InputObject <IVMWareIdentity> [-DefaultProfile <PSObject>] [-AsJob] [-NoWait]
+ [-PassThru] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Delete a private cloud
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Remove-AzVMWarePrivateCloud.md:32]: Here miss the examples.

File: src/VMWare/docs/Test-AzVMWareLocationQuotaAvailability.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+Test-AzVMWareLocationQuotaAvailability -InputObject <IVMWareIdentity> [-DefaultProfile <PSObject>] [-Confirm]
+ [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Return quota for subscription by region
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Test-AzVMWareLocationQuotaAvailability.md:32]: Here miss the examples.

File: src/VMWare/docs/Test-AzVMWareLocationTrialAvailability.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+Test-AzVMWareLocationTrialAvailability -InputObject <IVMWareIdentity> [-DefaultProfile <PSObject>] [-Confirm]
+ [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Return trial status for subscription by region
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Test-AzVMWareLocationTrialAvailability.md:32]: Here miss the examples.

File: src/VMWare/docs/Update-AzVMWareCluster.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+Update-AzVMWareCluster -InputObject <IVMWareIdentity> [-ClusterSize <Int32>] [-DefaultProfile <PSObject>]
+ [-AsJob] [-NoWait] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Update a cluster in a private cloud
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Update-AzVMWareCluster.md:33]: Here miss the examples.

File: src/VMWare/docs/Update-AzVMWarePrivateCloud.md [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+ [-Internet <InternetEnum>] [-ManagementClusterSize <Int32>] [-Tag <Hashtable>] [-DefaultProfile <PSObject>]
+ [-AsJob] [-NoWait] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Update a private cloud
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
```
Comment (wyunchi-ms) [inline src/VMWare/docs/Update-AzVMWarePrivateCloud.md:34]: Here miss the examples.

File: src/VMWare/.gitignore [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+internal
+exports
+custom/*.psm1
+test/*-TestResults.xml
+/*.ps1
+/*.ps1xml
+/*.psm1
+/*.snk
+/*.csproj
+/*.nuspec
```
Comment (isra-fel) [inline src/VMWare/.gitignore:14]: tools/ should be in .gitignore and should not be checked in -- Xiaogang

File: src/VMWare/tools/Resources/custom/New-AzDeployment.ps1 [hunk 9]
```diff
@@ -0,0 +1,231 @@
+function New-AzDeployment {
+    [OutputType('Microsoft.Azure.PowerShell.Cmdlets.Resources.Models.Api20180501.IDeploymentExtended')]
+    [CmdletBinding(DefaultParameterSetName='CreateWithTemplateFileParameterFile', PositionalBinding=$false, SupportsShouldProcess, ConfirmImpact='Medium')]
```
Comment (wyunchi-ms) [inline src/VMWare/tools/Resources/custom/New-AzDeployment.ps1:3]: Is this file need?

File: src/VMWare/readme.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+``` yaml
+require:
+# readme.azure.noprofile.md is the common configuration file
+  - $(this-folder)/../readme.azure.noprofile.md
+input-file:
+  - https://github.com/Azure/azure-rest-api-specs/blob/master/specification/vmware/resource-manager/Microsoft.AVS/stable/2020-03-20/vmware.json
+
+module-version: 0.1.0
+title: VmWare
```
Comment (isra-fel) [inline src/VMWare/readme.md:57]:
```suggestion
title: VMWare
```

File: src/VMWare/tools/Resources/.gitattributes [hunk 11]
```diff
@@ -0,0 +1 @@
+* text=auto
```
Comment (wyunchi-ms) [inline src/VMWare/tools/Resources/.gitattributes:1]: I'm not sure whether the folder `Resources` under tools is necessary.

File: src/VMWare/custom/New-AzVMWarePrivateCloud.ps1 [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+        [System.String]
+        # The block of addresses should be unique across VNet in your subscription as well as on-premise.
+        # Make sure the CIDR format is conformed to (A.B.C.D/X) where A,B,C,D are between 0 and 255, and X is between 0 and 22
+        ${NetworkBlock},
+    
+        [Parameter(Mandatory)]
+        [Microsoft.Azure.PowerShell.Cmdlets.VMWare.Category('Body')]
+        [System.String]
+        # The name of the SKU.
+        ${SkuName},
```
Comment (isra-fel) [inline src/VMWare/custom/New-AzVMWarePrivateCloud.ps1:87]: I believe we have a convention that all `SkuName` should be called `Sku`

**Review by isra-fel — CHANGES_REQUESTED** [review]

**Review by wyunchi-ms — APPROVED** [review]

**Review by wyunchi-ms — APPROVED** [review]

**Review by isra-fel — APPROVED** [review]

**Pull request merged**